wordpress-mobile/WordPress-iOS
# Core: Add support for remote configuration

Closes #19445
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/547

## Description
Adds support for remote config params in the apps. We make a call to the remote config endpoint on every launch and relaunch.
The PR does not introduce any visible changes.

## Pre-testing notes

1. Parsing the data returned from the endpoint currently fails, because the endpoint is returning an array when it should return a dictionary. This will be fixed in the future from the BE side. (Ref: p1665706579665139/1665668952.589099-slack-C04564QS0V7)
2. Testing instructions will require a network debugging app like [Charles](https://www.charlesproxy.com/) or [Proxyman](https://proxyman.io/). 

## Testing Instructions

1. Install the WordPress app
2. Launch the app
3. Make sure a call to `/remote-config` is made
4. Minimize the app
5. Open it again
6. Make sure another call to `/remote-config` is made

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Added `RemoteConfigParameterTests`

PR submission checklist:
